sopel/plugin.py

```diff
@@ -2025,3 +2028,8 @@ def add_attribute(function):
         function.output_prefix = prefix
         return function
     return add_attribute
+
+
+def abort(msg: str | None = None) -> None:
```

Comment:
When a function never returns (because it'll always raise an exception), you should use `typing.NoReturn` (see https://docs.python.org/3/library/typing.html#typing.NoReturn).

Reply:
Thanks! :blue_heart: